style/servo/selector_parser.rs

```diff
@@ -708,6 +724,30 @@ impl<'a, 'i> ::selectors::Parser<'i> for SelectorParser<'a> {
                 }
                 ServoTableWrapper
             },
+            "-servo-text-control-inner-editor" => {
+                if !self.in_user_agent_stylesheet() {
+                    return Err(location.new_custom_error(SelectorParseErrorKind::UnexpectedIdent(name.clone())))
+                }
+                ServoTextControlInnerEditor
+            },
+            "-servo-text-control-inner-container" => {
+                if !self.in_user_agent_stylesheet() {
+                    return Err(location.new_custom_error(SelectorParseErrorKind::UnexpectedIdent(name.clone())))
+                }
+                ServoTextControlInnerContainer
+            },
+            "-servo-text-control-placeholder" => {
```

Comment:
Is the plan to eventually expose these to author stylesheets?

Reply:
Sorry I was already offline yesterday. Yes, that is the plan, Ideally if we have a proper UA Shadow DOM (or equivalent structure) for `<input>` in Servo, the implementation could be short. But we are yet to do that, and I afraid it would bloat the implementation, so we are doing this step by step. But it does make sense to add the definition of the standardized version as well.

Reply:
After having more thoughts about this, It does make sense. I am thinking about reducing the dependency of changes later. But it would not make sense that we are exposing a false positive (i.e., passing the WPT for parsing `::placeholder` but are not actually implementing it).